lib/Transforms/SimplifyAffineStructures.cpp

```diff
@@ -110,8 +111,21 @@ void SimplifyAffineStructures::runOnFunction() {
   for (auto allocOp : allocOps) {
     normalizeMemRef(allocOp);
   }
+
+  // Remove zero trip count loops.
+  // TODO: templatize walk methods on operation interfaces.
+  // TODO: this could be moved to a more appropriate place.
+  func.walk([&](Operation *op) {
+    auto loopOp = dyn_cast<LoopLikeOpInterface>(op);
+    if (!loopOp)
+      return;
+    auto tripCount = loopOp.getConstantTripCount();
+    if (tripCount.hasValue() && tripCount.getValue() == 0)
```

Comment:
nit: On these compares you don't have to check for 'hasValue'. You can just compare directly: tripCount == 0